mirgecom/simutil.py

```diff
@@ -537,7 +1103,10 @@ def configurate(config_key, config_object=None, default_value=None):
         d = config_object if isinstance(config_object, dict) else\
             config_object.__dict__
         if config_key in d:
-            return d[config_key]
+            value = d[config_key]
+            if default_value is not None:
+                return type(default_value)(value)
+            return value
```

Comment:
I have a hunch that `configurate` can be replaced with [dict.get()](https://docs.python.org/3/library/stdtypes.html#dict.get), but it's fine to keep it around for now. I'll make an issue to remind me to look into it.

Reply:
https://github.com/illinois-ceesd/mirgecom/issues/879

Reply:
It is written to accept objects that are not necessarily straight dictionaries.   I have a refactor of the examples that pass a class here and it still works for those.  I'd like to retain that capability.

Reply:
Suspect there may be cleaner ways to do that vs. `configurate`. Do you have that refactor on Github anywhere yet where I could take a look?

Reply:
iirc, the changes were contained in a branch.  this one? (https://github.com/illinois-ceesd/mirgecom/tree/refactor-ux)  But it is *really* old.  I might do it different now, and would like the chance to.   I welcome the idea of doing this configure thing better, just not sure its worth it to go after right now (as you mentioned).  We've been using this and it works for our purposes.